services/store.go

```diff
@@ -181,71 +187,38 @@ outer:
 					if !ok {
 						continue
 					}
-					xfer := decodeTransfer(height, info, evt)
-					if xfer == nil {
-						log.Warnf(
-							"No transfer detected for state %#v in transaction %s at height %d",
-							evt.States, info.TxHash.ToHexString(), height,
-						)
-						continue
-					}
-					if failed {
-						if evt.ContractAddress != ongAddr {
+					//check evm ong event log
+					isEvm, eventLog := checkEvmEventLog(evt)
+					if isEvm {
+						xfer, err := parseEvmOngTransferLog(eventLog, s.parsedAbi, info.GasConsumed)
+						if err != nil {
+							log.Warnf("parse evm ong err:%s,height:%d,txhash:%s", err, height, info.TxHash.ToHexString())
 							continue
 						}
-						// NOTE(tav): We skip additional transfer events on
-						// failed transactions if we've already matched the gas
-						// fee.
-						if gasVerified {
+						gasverified, isgas, isContinue := checkgasVerified(evt.ContractAddress, ori.Payer, xfer.from, failed, gasVerified, xfer.isGas)
```

Comment:
Since this code is repeated for both EVM and non-EVM transactions, would it perhaps be clearer to have it in just one place after the `if isEVM` check?

Reply:
great, I have optimized，optimize code